Lib/test/support/__init__.py

```diff
@@ -2873,3 +2874,40 @@ def __iter__(self):
         if self.iter_raises:
             1/0
         return self
+
+
+def in_systemd_nspawn() -> bool:
+    """
+    Test whether the test suite is running inside of systemd-nspawn container
+    """
+
+    try:
+        with open("/run/systemd/container", "rb") as fp:
+            return fp.read().rstrip() == b"systemd-nspawn"
+    except FileNotFoundError:
+        return False
```

Comment:
Can you just inline this code in in_systemd_nspawn_sync_suppressed()? This function doesn't seem to be used outside in_systemd_nspawn_sync_suppressed().

Reply:
Sure can do. My idea was that having it separate makes it self-explanatory, and potentially useful if we ever have any other problems with `systemd-nspawn`, but I'm fine either way.